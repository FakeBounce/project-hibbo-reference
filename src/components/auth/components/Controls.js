import React from 'react';
import { Animated } from 'react-native';
import { shape, instanceOf, func } from 'prop-types';
import EStyleSheet from 'react-native-extended-stylesheet';
import styled from 'styled-components/native';

import Link from 'shared/Link';

import { getTranslations } from 'utils/i18n';

const styles = EStyleSheet.create({
  link: {
    backgroundColor: 'transparent',
  },
  register: {
    fontSize: 35,
    marginBottom: 15,
  },
  login: {
    fontSize: 17,
  },
  linkText: {
    color: '$colors.white',
    fontFamily: '$fonts.circularMedium',
  },
});

const AWrapperControls = styled(Animated.View)`
  flex: 1;
  align-items: center;
  justify-content: flex-end;
  margin-bottom: 50px;
`;

const Controls = ({ styling, selectAuthType }) => {
  return (
    <AWrapperControls style={styling}>
      <Link
        label={getTranslations('auth.register')}
        action={() => {
          selectAuthType('register');
        }}
        buttonStyle={styles.link}
        linkStyle={[styles.linkText, styles.register]}
      />

      <Link
        label={getTranslations('auth.login')}
        action={() => {
          selectAuthType('login');
        }}
        buttonStyle={styles.link}
        linkStyle={[styles.linkText, styles.login]}
      />
    </AWrapperControls>
  );
};

Controls.propTypes = {
  styling: shape({
    opacity: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  selectAuthType: func.isRequired,
};

export default Controls;
