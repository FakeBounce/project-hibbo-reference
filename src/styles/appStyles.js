/* eslint global-require: 0 */
import { Dimensions } from 'react-native';

import { iconVideos } from 'utils/storedData';

const { width } = Dimensions.get('window');

export default {
  prefix: {
    fixedIcon: 'icon_opac/',
  },
  event: {
    none: 0,
    gateau: 1,
    sapin: 2,
    oeuf: 3,
    souris: 4,
    bonbon: 11,
    manette: 12,
    moto: 13,
    voiture: 14,
    cadeau_vert: 15,
    cadeau_bleu: 16,
    courses: 21,
    rangement: 22,
    crayon: 23,
    marteau: 24,
    lapin: 31,
    chat: 32,
    chien: 33,
    heureux: 41,
    betise: 42,
    system: {
      reset: 0xffff,
      setLocal: 0xfff0,
      setVolume: 0xfff1,
      setObjectif: 0xfff2,
    },
  },
  icons: iconVideos,
  maxTextSize: 70,
  colors: {
    white: '#FFF',
    black: '#000',
    darkGrey: '#303030',
    darkBlue: '#06263D',
    darkBlueModal: '#06263D',
    grey: '#979797',
    greyOpac: 'rgba(151, 151, 151, 0.18)',
    whiteOpac: 'rgba(255, 255, 255, 0.30)',
    green: '#16EBC5',
    fb: '#4100C0',
    google: '#D5492F',
    primary: '#413D3D',
    textGrey: '#A7A7A7',
    buttonGrey: '#EFEFEF',
    lightGrey: '#9B9B9B',
    lightGreyBullet: '#E5E3DD',
    blueGreen: '#16EBC5',
    borderGrey: 'rgba(74, 74, 74, 0.23)',
    blackModal: 'rgba(0, 0, 0, 0.5)',
    toggleButtonGrey: '#4A4A4A',
    yellow: '#FFDF2D',
    moneyPotText: '#FFFE82',
    buttonYellow: '#FFFE82',
    cardBlue: '#697A99',
    gold: '#FEE9BF',
    red: '#FC5151',
    error: '#FF6969',
    textLink: '#687A99',
    messageGrey: '#DDE6EC',
    blue: '#0084ff',
    balanceGreen: '#3CDD80',
    pelicanColor: '#A685FF',
    deepBlue: '#028DAF',
    flatPink: '#FF6682',
    flatOrange: '#FFB434',
    flatPurple: '#82438F',
    white2pac: 'rgba(255,255,255,0.2)',
    white1pac: 'rgba(255,255,255,0.1)',
    placeholderGrey: 'rgba(229, 227, 221, 0.5)',
    greenStrong: '#50E3C2',
    orangeNotStrong: '#F5A623',
    inputLabel: '#C3C3C3',
    inputGrey: '#E8E8E8',
  },
  transaction: {
    minimunForAnimation: 1000,
  },
  time: {
    moneyTransaction: 2500,
  },
  textSizes: {
    xxxsmall: 10,
    xxsmall: 13,
    xsmall: 16,
    small: 18,
    regular: 22,
    normal: 24,
    xnormal: 26,
    big: 36,
    xbig: 50,
    xxbig: 58,
  },
  size: {
    cardMargin: 10,
    maxBubbleWidth: 190,
  },
  cardStyle: {
    shadowOpacity: 0,
    shadowColor: 'transparent',
    borderRadius: 10,
  },
  fonts: {
    circularBook: 'Circular-Std-Book',
    circularBookItalic: 'Circular-Std-Book-Italic',
    circularMedium: 'Circular-Std-Medium',
    circularMediumItalic: 'Circular-Std-Medium-Italic',
    circularBold: 'Circular-Std-Bold',
    circularBoldItalic: 'Circular-Std-Bold-Italic',
    circularBlackItalic: 'Circular-Std-Black-Italic',
    pixel: 'LCDDot-TR',
    chewy: 'Chewy',
    peepo: 'Peepo',
  },
  get colorList() {
    return [
      this.colors.deepBlue,
      this.colors.flatPink,
      this.colors.flatOrange,
      this.colors.flatPurple,
    ];
  },
  colorListSuffix: ['_bleu', '_rose', '_orange', '_violet'],
  swiperRange: [0, width / 2, width, width * 1.5, width * 2],
};
