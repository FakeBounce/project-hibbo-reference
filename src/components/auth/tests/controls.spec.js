import { Animated } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EStyleSheet from 'react-native-extended-stylesheet';

import appStyles from 'styles/appStyles';

import Controls from '../components/Controls';

EStyleSheet.build(appStyles);

describe('Controls', () => {
  const fn = jest.fn();
  const styling = {
    opacity: new Animated.Value(100),
  };

  it('should render the component', () => {
    const component = renderer.create(<Controls styling={styling} selectAuthType={fn} />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
