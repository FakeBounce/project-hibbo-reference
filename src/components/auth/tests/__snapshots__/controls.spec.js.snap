// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should render the component 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-end",
      "marginBottom": 50,
      "opacity": 100,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Circular-Std-Medium",
          },
          Object {
            "fontSize": 35,
            "marginBottom": 15,
          },
        ]
      }
    >
      Register
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Circular-Std-Medium",
          },
          Object {
            "fontSize": 17,
          },
        ]
      }
    >
      Login
    </Text>
  </View>
</View>
`;
